src/RoleObjects/ArticleData.test.js

```diff
@@ -0,0 +1,80 @@
+import * as createjs from 'createjs-module';
+import AccessibilityModule from '../index';
+
+describe('ArticleData', () => {
+  describe('register role', () => {
+    let canvasEl;
+    let parentEl;
+    let stage;
+    let container;
+    let cjsArticle;
+    let articleEl;
+    let positionVal;
+    let sizeVal;
+
+    beforeEach(() => {
+      canvasEl = document.createElement('canvas');
+      parentEl = document.createElement('div');
+      stage = new createjs.Stage(canvasEl);
+      container = new createjs.Container();
+      cjsArticle = new createjs.Shape(); // dummy object
+      positionVal = 7;
+      sizeVal = 99;
+
+      AccessibilityModule.register({
+        displayObject: container,
+        role: AccessibilityModule.ROLES.MAIN,
+      });
+      AccessibilityModule.setupStage(stage, parentEl);
+      stage.accessibilityTranslator.root = container;
+      stage.addChild(container);
+
+      AccessibilityModule.register({
+        accessibleOptions: {
+          position: positionVal,
+          size: sizeVal,
+        },
+        displayObject: cjsArticle,
+        parent: container,
+        role: AccessibilityModule.ROLES.ARTICLE,
+      });
+
+      stage.accessibilityTranslator.update();
+    });
+
+    describe('rendering', () => {
+      it('creates article element', () => {
+        articleEl = parentEl.querySelector('article');
+        expect(articleEl).not.toBeNull();
+      });
+
+      it('sets \'aria-position\' attribute', () => {
+        articleEl = parentEl.querySelector(`article[aria-posinset='${positionVal}']`);
+        expect(articleEl).not.toBeNull();
```

Comment:
Since the parentEl is recreated every test run (it's not being reused), I'm thinking about failure scenarios (when the tests don't pass), and thinking a better pattern is probably to test the actual attribute return value, rather than rely on the querySelector (the former will provide a more readable error/failure text in the case the test doesn't pass).

```suggestion
        articleEl = parentEl.querySelector(`article`);
        expect(articleEl.ariaPosInSet).toEqual(positionVal);
```

Reply:
@mmoss it barfed:
```  ● ArticleData › register role › rendering › sets "aria-position" attribute

    expect(received).toEqual(expected) // deep equality

    Expected: 7
    Received: undefined

      51 |       it('sets "aria-position" attribute', () => {
      52 |         articleEl = parentEl.querySelector('article');
    > 53 |         expect(articleEl.ariaPosInSet).toEqual(positionVal);
         |                                        ^
      54 |       });
      55 |
      56 |       it('sets "aria-setsize" attribute', () => {

      at Object.<anonymous> (src/RoleObjects/ArticleData.test.js:53:40)

Reply:
Hmmm... This makes me wonder if the test setup is doing what we expect (i.e. is there more than one article; do we need an `afterEach` to remove the children of the `parentEl`/from the `document`?

Reply:
when I log it, the attribute is there... for some reason it just doesn't like me using that property as a getter

even my IDE is flagging the code as an 'unresolved variable' ¯\\\_(ツ)_/¯

Reply:
Ugh. Looks like it's just too new (https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaPosInSet), maybe `getAttribute('aria-posinset')` will do the trick?

Reply:
yeah, that's the ONLY way I found to get it to work when testing the value itself (vs the selector)

unfortunately, now we have to cast to do the comparison 👎 

```    expect(received).toEqual(expected) // deep equality

    Expected: 7
    Received: "7"

      52 |         articleEl = parentEl.querySelector('article');
      53 |         console.log(articleEl.outerHTML);
    > 54 |         expect(articleEl.getAttribute('aria-posinset')).toEqual(positionVal);
         |                                                         ^
      55 |       });
      56 |
      57 |       it('sets "aria-setsize" attribute', () => {

Reply:
For the attributes that are fully supported, let's use the `getAttribute`, and I think the type conversion is likely only going to be an issue when the attribute isn't supported yet (it's in the spec, but browser hasn't implemented it), otherwise we _should_ get the proper value type.